### ISSUE TITLE
Collection item count discrepancy

### ISSUE THREAD

**Issue by alishaevn** [samvera/hyrax#5072]
### Descriptive summary
Hyrax `v.2.9.5`
The amount of works in a collection is showing one number on the dashboard index page for all collections, but a different number when clicking into that collection. I’m unable to duplicate this in local dev, but it’s happening on production in the app I'm working on. I tracked down the 2 methods that render both item counts. One appears to be calling active fedora, while the other is calling hyrax presenter renderer.

#### **collections index
[_list_collections.html.erb](https://github.com/samvera/hyrax/blob/v2.9.5/app/views/hyrax/dashboard/collections/_list_collections.html.erb)

``` ruby
# erb page
<td><%= collection_presenter.total_viewable_items %></td>

# method
def total_viewable_items do
  ActiveFedora::Base.where("isPartOf_ssim:#{id}").accessible_by(current_ability).count
end
```

<details> 
  <summary>this method gets the id of the admin set</summary>
  <pre>
    id >>> 9cbd202c-94b8-48ac-a345-01751fff1554 (admin 2)
  </pre>
</details>

<details> 
  <summary>then it goes into active fedora and gets the works whose `admin_set_id` match the id of the admin set</summary>
     where >>> [#<GenericWork id: "1dd38806-ca9e-4753-8532-e49a5631c8f0", head: [], tail: [], depositor: "admin@example.com", title: ["U Move, I Move"], date_uploaded: "2021-08-09 22:07:16", date_modified: "2021-08-09 22:07:16", state: #<ActiveTriples::Resource:0x25a058 ID:<http://fedora.info/definitions/1/0/access/ObjState#active>>, proxy_depositor: nil, on_behalf_of: nil, arkivo_checksum: nil, owner: nil, bulkrax_identifier: nil, institution: [], org_unit: [], refereed: nil, funder: [], fndr_project_ref: [], add_info: nil, date_published: nil, date_accepted: nil, date_submitted: nil, project_name: [], rights_holder: [], doi: nil, place_of_publication: [], abstract: nil, alternate_identifier: [], related_identifier: [], creator_search: ["Legend, John"], library_of_congress_classification: [], doi_options: nil, draft_doi: nil, disable_draft_doi: nil, alt_title: [], dewey: nil, file_availability: ["Filenot available"], collection_id: [], collection_names: [], volume: [], pagination: nil, issn: nil, eissn: nil, official_link: nil, series_name: [], edition: nil, event_title: [], event_date: [], event_location: [], book_title: nil, journal_title: nil, issue: nil, article_num: nil, isbn: nil, media: [], related_exhibition: [], related_exhibition_date: [], version: [], version_number: [], alternative_journal_title: [], related_exhibition_venue: [], current_he_institution: [], qualification_name: nil, qualification_level: nil, duration: [], editor: [], label: nil, relative_path: nil, import_url: nil, resource_type: ["GenericWork Musical notation"], creator: ["[{\"creator_given_name\":\"John\",\"creator_family_name\":\"Legend\",\"creator_name_type\":\"Personal\",\"creator_position\":\"1\"}]"], contributor: [], description: [], keyword: [], license: [], rights_statement: [], publisher: [], date_created: [], subject: [], language: [], identifier: [], based_near: [], related_url: [], bibliographic_citation: [], source: [], access_control_id: "f415b85a-4946-40e5-9bc4-5ba61fc694d1", representative_id: nil, thumbnail_id: nil, rendering_ids: [], admin_set_id: "9cbd202c-94b8-48ac-a345-01751fff1554", embargo_id: nil, lease_id: nil>]
</details>

<details> 
  <summary>next it filters those by ones that the current user has the ability to look at</summary>
  <pre>
    I'm signed in as an admin so it's the same work as above
  </pre>
</details>

<details> 
  <summary>finally, it returns the count the filtered works</summary>
  <pre>
    count >>> 1
  </pre>
</details>

#### ** collections show
[_show_descriptions.html.erb](https://github.com/samvera/hyrax/blob/v2.9.5/app/views/hyrax/dashboard/collections/_show_descriptions.html.erb)

``` ruby
# erb page
<h2 class="sr-only">Descriptions</h2>
<dl class="dl-horizontal metadata-collections">
  <% present_terms(@presenter, @presenter.terms_with_values) do |r, term| %>
      <dt><%= r.label(term) %></dt>
      <dd><%= r.value(term) %></dd>
  <% end %>
</dl>

# method
def present_terms(presenter, terms = :all, &block)
  Hyrax::PresenterRenderer.new(presenter, self).fields(terms, &block)
end
```

<details> 
  <summary>presenter argument</summary>
  <pre>
    presenter >>> admin 2
  </pre>
</details>

<details> 
  <summary>terms argument</summary>
  <pre>
    terms >>> [:total_items, :size, :creator]
  </pre>
</details>

<details> 
  <summary>Hyrax::PresenterRenderer.fields</summary>
  <pre>
    def fields(terms, &_block)
      @view_context.safe_join(terms.map { |term| yield self, term })
    end
  </pre>

> map >>> self: #<Hyrax::PresenterRenderer:0x0000559860728398>, term: total_items<br>
> map >>> self: #<Hyrax::PresenterRenderer:0x0000559860728398>, term: size<br>
> map >>> self: #<Hyrax::PresenterRenderer:0x0000559860728398>, term: creator<br>
  
</details>

### Rationale
These 2 numbers should be the same.

### Expected behavior
The item count on the index page should be _the same_ as the count on the show page.

### Actual behavior
The item count on the index page is _less than_ the count on the show page.
| index | show |
| --- | --- |
|  ![Screen Shot 2021-08-16 at 12 07 00 PM](https://user-images.githubusercontent.com/29032869/129616916-bc659a12-7778-4864-99aa-da69e0528cf6.png) | ![Screen Shot 2021-08-16 at 12 10 32 PM](https://user-images.githubusercontent.com/29032869/129616998-a4b59fe2-d001-49f9-be7e-2f54c1a84aee.png) |

### Steps to reproduce the behavior
I've been unable to reproduce this locally. My assumption though is that it has something to do with the way the 2 counts are being collected.

**Comment by alishaevn** [comment]
TLDR: tested this in `v.3.0.0` too but got stuck at valkyrie.

# index
[_list_collections.html.erb](https://github.com/samvera/hyrax/blob/b034218b89dde7df534e32d1e5ade9161e129a1d/app/views/hyrax/dashboard/collections/_list_collections.html.erb) in 3.0 calls the same `total_viewable_items` method. [that version of the method](https://github.com/samvera/hyrax/blob/e5fdb2b1caeb78ec3a9ecf4c186a0aad2ca5f0f2/app/presenters/hyrax/collection_presenter.rb#L87) is different though:
``` ruby
# erb page
<td><%= collection_presenter.total_viewable_items %></td>

# method
def total_viewable_items
  field_pairs = { "member_of_collection_ids_ssim" => id.to_s }
  SolrQueryService.new
                  .with_field_pairs(field_pairs: field_pairs)
                  .accessible_by(ability: current_ability)
                  .count
end
```

# show
there isn't an "app/views/hyrax/dashboard/collections/_show_descriptions.html.erb" file in 3.0. there is a [app/views/hyrax/collections/_show_descriptions.html.erb](https://github.com/samvera/hyrax/blob/e5fdb2b1caeb78ec3a9ecf4c186a0aad2ca5f0f2/app/views/hyrax/collections/_show_descriptions.html.erb) file though. (no "dashboard" folder). the erb page is calling a [terms_with_values](https://github.com/samvera/hyrax/blob/e5fdb2b1caeb78ec3a9ecf4c186a0aad2ca5f0f2/app/presenters/hyrax/collection_presenter.rb#L50) method instead:

``` ruby
# erb file
<% @presenter.terms_with_values.each do |field_name| %>
  <div id=<%= "hyc-collection-metadata-#{field_name}" %>>
    <dt><%= collection_metadata_label(@presenter, field_name) %></dt>
    <dd><%= collection_metadata_value(@presenter, field_name) %></dd>
  </div>
<% end %>

# method
def terms_with_values
  self.class.terms.select { |t| self[t].present? }
end
```


in 3.0 the erb files are calling 2 different methods, but both methods are defined in the CollectionPresenter.
updating the "show" erb page and both methods resulted in an `uninitialized constant Hyrax::CollectionPresenter::SolrQueryService` error. I added the [SolrQueryService](https://github.com/samvera/hyrax/blob/c211c40e3a673ca905947f36303714f73d175251/app/services/hyrax/solr_query_service.rb#L8) file.

that resulted in an `undefined method 'query_service' for Hyrax:Module` error. `query_service` is defined in 5 files. I chose the method in the [hyrax/lib/hyrax.rb](https://github.com/samvera/hyrax/blob/c211c40e3a673ca905947f36303714f73d175251/lib/hyrax.rb#L134) file. I can't get past that error though because that method calls another method that's using Valkyrie, which is not in `v.2.9.5`.
